lib/pbench/test/unit/server/test_server_configuration.py

```diff
@@ -0,0 +1,220 @@
+                headers={"authorization": f"bearer {token}"},
+                **kwargs,
+            )
+            assert response.status_code == expected_status
+            return response
+
+        return query_api
+
+    def test_get_bad_keys(self, query_get):
+        response = query_get("xyzzy", HTTPStatus.BAD_REQUEST)
+        assert response.json == {
+            "message": "Unrecognized keyword ['xyzzy'] given for parameter key; allowed keywords are ['dataset-lifetime', 'server-banner', 'server-state']"
+        }
+
+    def test_get1(self, query_get):
+        response = query_get("dataset-lifetime")
+        assert response.json == {"dataset-lifetime": "3650"}
+
+    def test_get_all(self, query_get):
+        response = query_get(None, HTTPStatus.OK)
+        assert response.json == {
+            "dataset-lifetime": "3650",
+            "server-state": {"status": "enabled"},
+            "server-banner": None,
+        }
+
+    def test_put_missing_key(self, query_put):
+        """
+        Test behavior when JSON payload does not contain all required keys.
+        """
+        response = query_put(json={}, key=None, expected_status=HTTPStatus.BAD_REQUEST)
+        assert response.json.get("message") == "Missing required parameters: config"
+
+    def test_put_bad_key(self, query_put):
+        response = query_put(key="fookey", expected_status=HTTPStatus.BAD_REQUEST)
+        assert response.json == {
+            "message": "Unrecognized keyword ['fookey'] given for parameter key; allowed keywords are ['dataset-lifetime', 'server-banner', 'server-state']"
+        }
+
+    def test_put_bad_keys(self, query_put):
+        response = query_put(
+            key=None,
+            json={"config": {"fookey": "bar"}},
+            expected_status=HTTPStatus.BAD_REQUEST,
+        )
+        assert response.json == {
+            "message": "Unrecognized JSON key ['fookey'] given for parameter config; allowed keywords are ['dataset-lifetime', 'server-banner', 'server-state']"
+        }
+
+    @pytest.mark.parametrize(
+        "key,value",
+        [
+            ("dataset-lifetime", "14 years"),
+            ("server-banner", {"banner": None}),
+            ("server-state", "running"),
+            ("server-state", {"status": "disabled"}),
+        ],
+    )
+    def test_bad_value(self, query_put, key, value):
+        response = query_put(
+            key=key, expected_status=HTTPStatus.BAD_REQUEST, json={"value": value}
+        )
+        assert (
+            f"Unsupported value for configuration key '{key}'"
+            in response.json["message"]
+        )
+
+    def test_put_param(self, query_put):
+        response = query_put(key="dataset-lifetime", query_string={"value": "2"})
+        assert response.json == {"dataset-lifetime": "2"}
+
+    def test_put_value(self, query_put):
+        response = query_put(key="dataset-lifetime", json={"value": "2"})
+        assert response.json == {"dataset-lifetime": "2"}
```

Comment:
Shouldn't these be parameterized, _a la_ `test_bad_value()`, so that we test all three keys, a variety of valid `"dataset-lifetime"` values (integers, strings with just numbers of various lengths, strings with numbers and `"day"`, strings with numbers and `"days"`, various amounts of leading, trailing, and interstitial whitespace), and a variety of valid `"server-state"` values (`"status"` set to each of its three valid values (in upper-, lower-, and mixed-case), with and without extra keys).

Reply:
At the API layer I'm more interested in testing that DB-layer errors are properly passed through. There's more exhausting good-cop/bad-cop testing at the DB layer.